Angular 19 update

ceo991 opened the pull request:
The ng-cli-packagr-tasks package is updated also to be able to upgrade to Angular 19

Commit:
upgrading nx and angular to 18

Commit:
Refactoring components to work with version 18

Commit:
typescript fix

Commit:
temporary fix on deeppurple-amber.scss

Commit:
Initial logic moved to ngOnInit()

Commit:
MatTooltip injection is fixed for Angular 18

Commit:
Cdk table _rowOutlet is fixed

Commit:
Meta row fix

Commit:
element issue is fixed

Commit:
trying to make context work

Commit:
Fixed context issues and added an bash script for replacing dist with node_modulse/@pabula

ceo991 committed:
Merge pull request #2 from big-cube/angular_18_update

Angular 18 update

Commit:
angular 19 first commit

david-moldenhauer committed:
upgrade

Commit:
sass compiler fixes

Commit:
removed replace_pebula from scripts

Commit:
injection context problem is fixed

Commit:
context error is fixed

Commit:
cell tooltip positioning issue is fixed

ceo991 committed:
Merge pull request #3 from big-cube/angular_19_upgrade

Angular 19 upgrade

Commit:
ngrid cell context fix

david-moldenhauer committed:
update row on selection change

Commit:
cell select view update issue is fixed

Commit:
cache issue is fixed

Commit:
Commit d69b4355: detach  and hasPlugin fix

Commit:
removed unnecessary console logs

shlomiassaf commented:
Thank you!
In the past month we've been migrating it internally, however we migrate in incremental steps (15,16,17 ...)
This is due to production considerations as we have a long chain to support.

We will review it and adjust, making sure to test as well.

Thank you!

shlomiassaf commented:
@ceo991 It appears that the documentation app was not migrated as well as the gh-pages script removed.

We will fix those internally, thank you!

ceo991 commented:
@shlomiassaf thank you so much for the reply. I can explain the changes which we have applied if you need. We have upgraded from 14 to 18 in one go then 18 to 19. Caching mechanisms were the main parts which we have struggled a lot btw.  We have upgraded ng-cli-packagr-tasks package to Angular 19 also there is an active PR. We are happy to contribute to the project. Thank you so much have a nice day.

RajvirSingh1313 commented:
@ceo991 This is amazing work you guys have done. I was having the same issue, was migrating a old project from angular 14 to 19 incrementally. And found that ngrid is now no longer maintained, and due to project using ngrid in a complex manner, it's looking like a big job for me to do.

Your work is really helpful, But wouldn't it be better if you guys make a npm package of it so others can too use it?

shlomiassaf commented:
@RajvirSingh1313 As I mentioned above, my team is working on the upgrade.
Once we have it working we'll release it.

RajvirSingh1313 commented:
@shlomiassaf Thanks alot, its very reassuring that you are working on it. But while I am on a client project, The deadline is near and this project is a big part of it. I have published a package from the guys above to make a temporary fix, but they havent really modified the ngrid-materials plugin.

So could just release like the upto the version you already have completed?
I know its asking alot, I am grateful that you are giving your time to such project.

Maybe I guess I will make the changes myself accordingly. Thanks btw for the work

Commit:
row cdref mark for check fix

david-moldenhauer committed:
fixed grouping header issue

david-moldenhauer committed:
element is defined in base

david-moldenhauer committed:
debug-logs and fixes

david-moldenhauer committed:
logs removed

david-moldenhauer committed:
invalidate columns (reorder fix)

Commit:
Update to angular 20

Hopefully these changes will also work with Angular 21 and 22.
With Angular 23 it is planned to remove the old structural directives.

Commit:
Fix sticky header

Partially reimplement the cdk sticky behavior

Commit:
Fix column auto resize

MaximilianLackaw committed:
Merge pull request #4 from big-cube/feature/angular20

Angular 20 update
